core/Plugin.php

```diff
@@ -359,7 +360,7 @@ public function findComponent($componentName, $expectedSubclass)
             $this->cache->save($cacheId, $classname);
         }
 
-        return new $classname;
+        return StaticContainer::get($classname);
```

Comment:
Don't know if you want to do this in this commit or not, but could also make this sort of change to callers of findMultipleComponents().